### PR TITLE
fix: deprecate tag logic on Association

### PR DESCRIPTION
**Pull request by danabens**
associations don't have arns so this tagging logic doesn't make sense

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/lineage/association.py
Comment (shreyapandit) [inline src/sagemaker/lineage/association.py:68]: > associations don't have arns so this tagging logic doesn't make sense

Looking at the [description](https://github.com/danabens/sagemaker-python-sdk/blob/remove-set-tag/src/sagemaker/lineage/association.py#L33) for this class, an association object still has a source and destination arn as attributes. Has this behavior changed?

Also, can we deprecate this instead and communicate this change instead of deleting it first, and then delete it eventually ? I am concerned with downstream consumers of the lineage APIs, who might be currently setting tags as part of their workflows and might find those workflows failing when they pull in the release with this change. Thanks so much!

Comment (danabens) [reply src/sagemaker/lineage/association.py:68]: > Has this behavior changed?

No, association does have  a source and dest but does not have an arn itself so it never made sense to have tagging behavior on this class in the first place. Clients tag the source and dest in other classes or by calling add_tags directly.

> can we deprecate this instead 

Ya, good idea.

**Review by jerrypeng7773 — APPROVED** [review]

**Review by ahsan-z-khan — APPROVED** [review]

File: src/sagemaker/lineage/association.py
Comment (shreyapandit) [reply src/sagemaker/lineage/association.py]: Thanks for the clarification! Waiting for tests to pass before merge

**Review by shreyapandit — APPROVED** [review]

**Pull request merged**
